definitions/scenarios/self_upgrade.rb

```diff
@@ -58,15 +66,37 @@ class SelfUpgrade < SelfUpgradeBase
       manual_detection
     end
 
-    def compose
+    def downstream_self_upgrade(_pkgs)
       if check_min_version('foreman', '2.5') || check_min_version('foreman-proxy', '2.5')
-        pkgs_to_update = %w[satellite-maintain rubygem-foreman_maintain]
         yum_options = req_repos_to_update_pkgs.map do |id|
           "--enablerepo=#{id}"
         end
         add_step(Procedures::Packages::Update.new(packages: pkgs_to_update, assumeyes: true,
```

Comment:
shouldn't this use `_pkgs` now? or is it using `pkgs_to_update` from `compose`? seems odd.